Add support for .norm() pytorch onnx export and ReduceL1/ReduceL2 caffe2 operators

fuzic opened the pull request:
Onnx has ReduceL1 and ReduceL2 operators that would facilitate this, so allow pytorch to export those and allow caffe2 to run them.

I only implemented this on CPU so far.

soumith commented:
cc: @bddppq 
@fuzic maybe needs a test?

houseroad reviewed (changes requested):
I don't think this is the right solution.

Please check https://github.com/onnx/onnx/blob/master/docs/Operators.md#reducel1 and https://github.com/onnx/onnx/blob/master/docs/Operators.md#reducel2

Commit:
Add support for .norm() pytorch onnx export and ReduceL1/ReduceL2 caffe2 operators (#9299)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/9299

Onnx has ReduceL1 and ReduceL2 operators that would facilitate this, so allow pytorch to export those and allow caffe2 to run them.

I only implemented this on CPU so far.

Differential Revision: D8757381

fbshipit-source-id: 040c5b317d65de4d246f755e224e2ef0cebfc578
